cognite/client/_api/simulators/models.py

```diff
@@ -0,0 +1,389 @@
+    CreatedTimeSort,
+    SimulatorModel,
+    SimulatorModelCore,
+    SimulatorModelList,
+    SimulatorModelRevision,
+    SimulatorModelRevisionCore,
+    SimulatorModelRevisionList,
+    SimulatorModelRevisionWrite,
+    SimulatorModelUpdate,
+    SimulatorModelWrite,
+)
+from cognite.client.utils._experimental import FeaturePreviewWarning
+from cognite.client.utils._identifier import IdentifierSequence
+from cognite.client.utils._validation import assert_type
+from cognite.client.utils.useful_types import SequenceNotStr
+
+if TYPE_CHECKING:
+    from cognite.client import ClientConfig, CogniteClient
+
+
+class SimulatorModelRevisionsAPI(APIClient):
+    _RESOURCE_PATH = "/simulators/models/revisions"
+
+    def __init__(self, config: ClientConfig, api_version: str | None, cognite_client: CogniteClient) -> None:
+        super().__init__(config, api_version, cognite_client)
+        self._warning = FeaturePreviewWarning(
+            api_maturity="General Availability", sdk_maturity="alpha", feature_name="Simulators"
+        )
+
+    def list(
+        self,
+        limit: int = DEFAULT_LIMIT_READ,
+        sort: CreatedTimeSort | None = None,
+        filter: SimulatorModelRevisionsFilter | dict[str, Any] | None = None,
+    ) -> SimulatorModelRevisionList:
+        """`Filter simulator model revisions <https://developer.cognite.com/api#tag/Simulator-Models/operation/filter_simulator_model_revisions_simulators_models_revisions_list_post>`_
+        Retrieves a list of simulator model revisions that match the given criteria
+        Args:
+            limit (int): Maximum number of results to return. Defaults to 10. Set to -1, float(“inf”) or None to return all items. sort (CreatedTimeSort | None): Sort order for the results.
+            sort (CreatedTimeSort | None): The criteria to sort by.
+            filter (SimulatorModelRevisionsFilter | dict[str, Any] | None): Filter to apply.
+        Returns:
+            SimulatorModelRevisionList: List of simulator model revisions
+        Examples:
+            List simulator model revisions:
+                            >>> from cognite.client import CogniteClient
+                >>> client = CogniteClient()
+                >>> res = client.simulators.models.revisions.list()
+        """
+        self._warning.warn()
+        return self._list(
+            method="POST",
+            limit=limit,
+            resource_cls=SimulatorModelRevision,
+            list_cls=SimulatorModelRevisionList,
+            sort=[CreatedTimeSort.load(sort).dump()] if sort else None,
+            filter=filter.dump(camel_case=True) if isinstance(filter, CogniteFilter) else filter,
+        )
+
+    @overload
+    def retrieve(
+        self,
+        external_id: str,
+        id: int | None = None,
+    ) -> SimulatorModelRevision | None: ...
+
+    @overload
+    def retrieve(self, external_id: SequenceNotStr[str], id: int | None = None) -> SimulatorModelRevisionList: ...
```

Comment:
Any reason why we support only a list of External IDs and not a list of IDs?

Reply:
We have just followed how most of the other resources have implemented this function. Do you think our users would have need for retrieving by multiple ids?

Reply:
We have just followed how most of the other resources have implemented this function. Do you think our users would have need for retrieving by multiple ids?

Reply:
No, I was just curious. I checked the other endpoints in the SDK and most implement both support for multiple Ids and ExternalIds, but I don't have a strong opinion, as I never needed to fetch multiple by Id. As long as it's documented it should be fine.

Reply:
either both arrays, of both single items, strange that we have it mixed? one is array and one is not

<img width="878" alt="Screenshot 2025-01-20 at 15 11 15" src="https://github.com/user-attachments/assets/2498d44c-3b31-4847-9a07-82c62ecca052" />
<img width="611" alt="Screenshot 2025-01-20 at 15 11 05" src="https://github.com/user-attachments/assets/2d128fd4-ec02-4149-83b4-ee3c4ffc6ef8" />

Reply:
Yeah, dumb me, just took it from the last PR we had created had and assumed all other resources were doing the same

Reply:
It's nice to have both retrieve and retrieve_multiple for all resources, as the user can avoid writing for loops and making unnecessary requests.

Reply:
done, ive modified both our resources to have both `retrieve` and `retrieve_multiple` following the assets resource.